### ISSUE TITLE
Unable to link certificate with special characters in the subject

### ISSUE THREAD

**Issue by vokac** [indigo-iam/iam#533]
I'm not able to link certificate with [this subject](https://vokac.web.cern.ch/vokac/tmp/user-tcs-ctu.pem) using IAM web interface that shows following dialog ![Link user certificate dialog](https://vokac.web.cern.ch/vokac/tmp/IAM_link_cert_error.png)
but nothing gets added by pushing "Link" button. It is also interesting that login page displays certificate subject differently ![Login dialog with user certificate subject](https://vokac.web.cern.ch/vokac/tmp/IAM_login_cert_details.png).

I'm not even able to add this certificate subject directly with API using
```
> PATCH /scim/Users/266d17cd-3f59-6093-b3ab-cc65afa33850 HTTP/1.1
> Host: atlas-auth.web.cern.ch
> Accept-Encoding: identity
> Content-type: application/scim+json
> Authorization: Bearer secret
> Content-Length: 582
>
> {"schemas": ["urn:ietf:params:scim:api:messages:2.0:PatchOp"], "operations": [{"op": "add", "path": "certificates", "value": {"urn:indigo-dc:scim:schemas:IndigoUser": {"certificates": [{"label": "cert-1", "subjectDn": "emailAddress=petr.vokac@fjfi.cvut.cz,CN=Petr Vok\\\\C3\\\\A1\\\\C4\\\\8D,C=CZ,L=Praha 6 - Dejvice,ST=Praha\\\\, Hlavn\\\\C3\\\\AD m\\\\C4\\\\9Bsto,street=Jugosl\\\\C3\\\\A1vsk\\\\C3\\\\BDch partyz\\\\C3\\\\A1n\\\\C5\\\\AF 1580/3,O=\\\\C4\\\\8Cesk\\\\C3\\\\A9 vysok\\\\C3\\\\A9 u\\\\C4\\\\8Den\\\\C3\\\\AD technick\\\\C3\\\\A9 v Praze,postalCode=160 00", "issuerDn": "CN=GEANT Personal CA 4,O=GEANT Vereniging,C=NL"}]}}}]}

< HTTP/1.1 400 Bad Request
< Server: nginx/1.17.9
< Date: Thu, 24 Nov 2022 23:44:27 GMT
< Content-Type: application/json;charset=UTF-8
< Transfer-Encoding: chunked
< Connection: keep-alive
< Access-Control-Allow-Origin: *
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< Strict-Transport-Security: max-age=31536000 ; includeSubDomains
< X-Frame-Options: DENY
< X-Application-Context: INDIGO IAM:mysql,flyway-repair,prod,oidc,cern,registration,wlcg-scopes:8080
```

**Comment by giacomini** [comment]
Are non-ascii characters allowed? see https://wiki.geant.org/display/TCSNT/TCS+2020+FAQ#TCS2020FAQ-Q:HowdoIdealwithnon-ASCIIcharactersforIGTFcertificates?

**Comment by giacomini** [comment]
I've found this https://ggus.eu/index.php?mode=ticket_info&ticket_id=152981. In particular, update#5 cites https://www.ogf.org/documents/GFD.125.pdf as forbidding non-ASCII characters in IGTF certificates. I'm checking if that document is still relevant.

**Comment by msalle** [comment]
Hi all,
the problem is that it's GEANT Personal CA 4 certificate, not the IGTF-MICS version (you should be able to choose that when getting a certificate from the GEANT TCS). For the personal UTF-8 is fine, for IGTF it's not. The best reference is https://www.ogf.org/documents/GFD.225.pdf as referred to from https://www.eugridpma.org/guidelines/pkitech/
See in particular §4.3 in GFD.225

**Comment by vokac** [comment]
"solved" by secondary CERN account which can be used to issue different certificate for my second IAM account.

Our national CA operator CESNET no longer [issue TCS grid certificates](https://pki.cesnet.cz/en/tcs-personal.html) and it is too complicated / time consuming to get personal certificate from their own CESNET CA4 personal authority.

**Comment by vokac** [comment]
Still, it would be nice if IAM prints error while trying to link "bad" certificate and not just blank page.

**Comment by msalle** [comment]
I agree a clear error message would definitely be good. Also, if it is IGTF only, it should not have the GEANT Personal CA 4 configured as acceptable CA, but only the IGTF bundles.

**Comment by giacomini** [comment]
From the output of `openssl s_client`, I see only IGTF CAs. I agree that the messages should be more informative.

**Comment by msalle** [comment]
Ah, right. The client is probably sending both the EEC and the "GEANT Personal CA 4" certificates. The latter (see https://crt.sh/?caid=160144) is issued by the same CA as the "GEANT eScience Personal CA 4" (see https://crt.sh/?caid=160134), hence the chain is still accepted since that common parent is in the IGTF distribution (ca_USERTrustRSACertificationAuthority-1.117-1). There isn't much one can do about that.
